webapp/channels/src/utils/message_html_to_component.tsx

```diff
@@ -0,0 +1,269 @@
+ * - mentionHighlight - If specified, mentions for the current user are highlighted. Defaults to true.
+ * - disableGroupHighlight - If specified, group mentions are not displayed as blue links. Defaults to false.
+ * - emoji - If specified, emoji text is replaced with the PostEmoji component. Defaults to true.
+ * - images - If specified, markdown images are replaced with the image component. Defaults to true.
+ * - imageProps - If specified, any extra props that should be passed into the image component.
+ * - latex - If specified, latex is replaced with the LatexBlock component. Defaults to true.
+ * - imagesMetadata - the dimensions of the image as retrieved from post.metadata.images.
+ * - hasPluginTooltips - If specified, the LinkTooltip component is placed inside links. Defaults to false.
+ * - channelId = If specified, to be passed along to ProfilePopover via AtMention
+ */
+
+type ShouldProcessNode = {type: string; name: string; attribs: {type: string; 'data-edited-post-id': string; href: string; 'data-sum-of-members-mention': string; 'data-mention': string; 'data-plan-mention': Record<string, string>; 'data-emoticon': string; 'data-latex': string; 'data-inline-latex': string; 'data-codeblock-code': string}}
+type ProcessNode = {attribs: {checked: string ; href: string; 'data-sum-of-members-mention': string; 'data-mention': string; 'data-plan-mention': string; [x: string]: string; src: string}; parentNode: {
+    type: string;
+    name: string;
+};}
+
+type ProcessingInstructions = {
+    replaceChildren?: boolean;
+    shouldProcessNode: (node: ShouldProcessNode) => boolean | string | Record<string, string>;
+    processNode: (node: ProcessNode, children?: boolean | ReactChild | ReactFragment | ReactPortal | null | undefined) => ReactElement<{
+        checked?: string | boolean;
+    }> | null;
+}
+
+export function messageHtmlToComponent(html: string, options: {postId?: string; editedAt?: number; hasPluginTooltips?: boolean; channelId?: string; atSumOfMembersMentions?: boolean; userIds?: string[]; messageMetadata?: Record<string, string>; atPlanMentions?: boolean; imagesMetadata?: Record<string, PostImage>; imageProps?: object; postType?: PostType; mentions?: boolean; mentionHighlight?: boolean; disableGroupHighlight?: boolean; emoji?: string; images?: string; latex?: string; inlinelatex?: string; markdown?: boolean}, isRHS?: boolean) {
+    if (!html) {
+        return null;
+    }
+
+    const parser = new Parser();
+    const processNodeDefinitions = new ProcessNodeDefinitions(React);
+
+    function isValidNode() {
+        return true;
+    }
+
+    const processingInstructions: ProcessingInstructions[] = [
+
+        // Workaround to fix MM-14931
+        {
+            replaceChildren: false,
+            shouldProcessNode: (node) => node.type === 'tag' && node.name === 'input' && node.attribs.type === 'checkbox',
+            processNode: (node) => {
+                let newAttribs = JSON.parse(JSON.stringify(node.attribs));
```

Comment:
Compared to old version, I believe we are changing the behaviour here. If OK for you, we can keep the old version